cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1228,6 +1480,60 @@ def set_dissimilarity_map(self, dissimilarity_map: pd.DataFrame) -> None:
 
         self.__dissimilarity_map = dissimilarity_map.copy()
 
+    def get_dissimilarity(self, node: str) -> Dict[str, float]:
+        """Get the dissimilarity of a single leaf to all other leaves.
+
+        Args:
+            node: Name of the leaf node to get the dissimilarity to/from
+
+        Raises:
+            CassiopeiaTreeError if the dissimilarity map is not defined or the
+                provided node is not in the dissimilarity map
+        """
+        if self.__dissimilarity_map is None:
+            raise CassiopeiaTreeError("No dissimilarity map is defined.")
+        if node not in self.__dissimilarity_map.index:
+            raise CassiopeiaTreeError(
+                f"`{node}` is not in the dissimilarity map."
+            )
+        return self.__dissimilarity_map.loc[node].to_dict()
+
+    def set_dissimilarity(
```

Comment:
For example, when a new leaf is added and we want to (partially) update the dissimilarity map, instead of re-computing the entire thing. In this case, because `add_leaf` calls `__register_data_with_tree`, which also adds the leaf (with infinite dissimilarity to all other nodes), we need to be able to overwrite existing dissimilarities to be able to set the correct values for this leaf after calling `add_leaf`.